### ISSUE TITLE
Tarantool segfaults after building index

### ISSUE THREAD

**Issue by ochaton** [tarantool/tarantool#6147]
**Bug description**

* OS: CentOS7
* Architecture: x86_64
* 1.10.8-36

**Steps to reproduce**
No steps to reproduce. Caught in production after building index under load.
Approximate DDL:

```lua
box.space.myspace:format({})
box.space.myspace:create_index('expiration', { { parts  = { 9,  "unsigned", is_nullable = true }, { 1, "string" }  } })
box.space.myspace:format({ ... })
```

**Actual behavior**

```
(gdb) bt
#0  0x00007fec9aeac387 in raise () from /lib64/libc.so.6
#1  0x00007fec9aeada78 in abort () from /lib64/libc.so.6
#2  0x0000000000417da3 in sig_fatal_cb (signo=<optimized out>, siginfo=<optimized out>, context=<optimized out>) at /build/usr/src/debug/tarantool-1.10.8.36/src/main.cc:258
#3  <signal handler called>
#4  0x000000000042aced in tuple_compare_with_key (key_def=0x2000001408b10, part_count=1, key=0x40845a71 <incomplete sequence \316>, tuple=0x7fe9bd34ca94)
    at /build/usr/src/debug/tarantool-1.10.8.36/src/box/key_def.h:778
#5  memtx_tree_compare_key (key_data=0x7fe89c982330, key_data=0x7fe89c982330, def=0x2000001408b10, tuple=0x7fe9bd34ca94) at /build/usr/src/debug/tarantool-1.10.8.36/src/box/memtx_tree.h:71
#6  tree_iterator_prev_equal (iterator=<optimized out>, ret=0x417c4fd0) at /build/usr/src/debug/tarantool-1.10.8.36/src/box/memtx_tree.c:187
#7  0x0000000000423705 in box_iterator_next (itr=<optimized out>, result=0x417c4fd0) at /build/usr/src/debug/tarantool-1.10.8.36/src/box/index.cc:401
#8  0x000000000051ba56 in lj_vm_ffi_call () at buildvm_x86.dasc:2576
#9  0x000000000055eca8 in lj_ccall_func (L=L@entry=0x4082ac98, cd=<optimized out>) at lj_ccall.c:1150
#10 0x000000000057365b in lj_cf_ffi_meta___call (L=0x4082ac98) at lib_ffi.c:230
#11 0x0000000000519aa7 in lj_BC_FUNCC () at buildvm_x86.dasc:809
#12 0x000000000052b472 in lua_pcall (L=L@entry=0x4082ac98, nargs=<optimized out>, nresults=<optimized out>, errfunc=errfunc@entry=0) at lj_api.c:1158
#13 0x00000000004ec9c3 in luaT_call (L=0x4082ac98, nargs=<optimized out>, nreturns=<optimized out>) at /build/usr/src/debug/tarantool-1.10.8.36/src/lua/utils.c:988
#14 0x00000000004e6be5 in lua_fiber_run_f (ap=<error reading variable: value has been optimized out>) at /build/usr/src/debug/tarantool-1.10.8.36/src/lua/fiber.c:371
#15 0x00000000004172ac in fiber_cxx_invoke(int (*)(__va_list_tag*), __va_list_tag*) () at /build/usr/src/debug/tarantool-1.10.8.36/src/fiber.h:760
#16 0x00000000004fe2b0 in fiber_loop (data=<optimized out>) at /build/usr/src/debug/tarantool-1.10.8.36/src/fiber.c:740
#17 0x0000000000686e4f in coro_init () at /build/usr/src/debug/tarantool-1.10.8.36/third_party/coro/coro.c:110
```

**Comment by Mjolkhare** [comment]
**Reproducer:**

```lua
box.cfg{}

box.schema.create_space("links", nil)
box.space.links:format({ {
	name = "id",
	type = "string"
  }, {
	name = "uid",
	type = "unsigned"
  }, {
	name = "fdir",
	type = "string"
  }, {
	name = "fname",
	type = "string"
  }, {
	name = "hash",
	type = "string"
  }, {
	name = "ctime",
	type = "unsigned"
  }, {
	name = "flags",
	type = "unsigned"
  } })
box.space.links:create_index("primary", {
  parts = { { 1, "string",
	  is_nullable = false
	} },
  type = "tree"
})
box.space.links:create_index("uid_ctime", {
  parts = { { 2, "unsigned",
	  is_nullable = false
	}, { 6, "unsigned",
	  is_nullable = false
	}, { 1, "string",
	  is_nullable = false
	} },
  type = "tree"
})

for i=1,10000 do
	box.space.links:insert(box.tuple.new{
		tostring(i),
		1,
		'test'..tostring(i),
		'test'..tostring(i),
		tostring(i),
		os.time(),
		0,
	})
end

local fiber = require 'fiber'
fiber.create(function()
	while true do
		box.space.links.index.uid_ctime:pairs(1):take_n(10):grep(function() fiber.sleep(0.001) return true end):length()
	end
end)
fiber.create(function() box.space.links:format({}) end)
```

**Comment by Korablev77** [comment]
Closed in 390c614a3b20845d0a0adb232aaa37fcd0401029
